core/src/test/scala/com/rackspace/com/papi/components/checker/GivenAWadlWithAuthenticatedByBase.scala

```diff
@@ -0,0 +1,535 @@
+    }
+
+  def createConfig(useSaxon: Boolean): Config = {
+    val config = new Config
+    config.resultHandler = TestConfig.assertHandler
+
+    if (useSaxon) {
+      config.xsdEngine                    = "SaxonEE"
+    }
+
+    // these are always true
+    config.checkWellFormed                = true
+    config.checkElements                  = true
+    config.checkPlainParams               = true
+    config.checkHeaders                   = true
+    config.enableAuthenticatedByExtension = true
+    config.enableAnyMatchExtension        = true
+
+    // these will be turned on for individual tests
+    config.enableRaxRolesExtension        = false
+    config.maskRaxRoles403                = false
+    config.removeDups                     = false
+    config.joinXPathChecks                = false
+    config.enableCaptureHeaderExtension   = false
+
+    config
+  }
+
+  def base: BaseValidatorSuite = new BaseValidatorSuite
+
+  def localWADLURI = new File(System.getProperty("user.dir"), "mywadl.wadl").toURI.toString
+
+  def response: HttpServletResponse = mock(classOf[HttpServletResponse])
+
+  def chain: FilterChain = mock(classOf[FilterChain])
+}
+
+object GivenAWadlWithAuthenticatedByBase {
+  val DefaultMediaType = "application/xml"
+  val DefaultRequest =
+    <request>
+      <representation mediaType={ DefaultMediaType }/>
+    </request>
+  val AuthByHeader = "X-Authenticated-By"
+  val RaxRolesHeader = "X-Roles"
+  val DeviceIdHeader = "X-Device-Id"
+  val ValidAuthenticatedByValues = Set("PASSWORD", "APIKEY", "PASSCODE", "OTPPASSCODE", "IMPERSONATION", "RSAKEY", "FEDERATED")
+  val InvalidAuthBy = List("POTATO")
+  val NoAuthBy = List()
+  val InvalidRaxRoles = roles("POTATO")
+  val NoRaxRoles = Map.empty[String, List[String]]
+  val WadlMethods = Set("GET", "POST", "PUT", "DELETE")
+  val HttpCodeBadRequest = 400
+  val HttpCodeUnauthorized = 401
+  val HttpCodeForbidden = 403
+  val HttpCodeNotFound = 404
+
+  type AccessExpectation = (String, String, List[String], String, Map[String, List[String]]) => Unit
+
+  type ConfigEnabledOption = Config => Unit
+
+  val EnableRaxRoles: ConfigEnabledOption = _.enableRaxRolesExtension = true
+  val EnableMaskRaxRoles403: ConfigEnabledOption = _.maskRaxRoles403 = true
+  val EnableRemoveDups: ConfigEnabledOption = _.removeDups = true
+  val EnableJoinXPathChecks: ConfigEnabledOption = _.joinXPathChecks = true
+  val EnableCaptureHeadersExt: ConfigEnabledOption = _.enableCaptureHeaderExtension = true
+
+  // all of the config option combinations we want to test with
+  val ConfigOptionCombinations: List[List[ConfigEnabledOption]] = List(
+    List(),
+    List(EnableRaxRoles),
+    List(EnableCaptureHeadersExt),
+    List(EnableRaxRoles, EnableCaptureHeadersExt),
+    List(EnableRaxRoles, EnableMaskRaxRoles403, EnableCaptureHeadersExt),
+    List(EnableRaxRoles, EnableMaskRaxRoles403, EnableRemoveDups, EnableJoinXPathChecks, EnableCaptureHeadersExt))
```

Comment:
ConfigOptionCombinations is the list of different configs I'm testing.  For config options that are always on, they are listed in the createConfig method on line 450.